newrelic/common/package_version_utils.py

```diff
@@ -31,40 +31,68 @@ def get_package_version(name):
                 "1.1.0"
     """
 
-    def _get_package_version(name):
-        module = sys.modules.get(name, None)
-        version = None
-        for attr in VERSION_ATTRS:
-            try:
-                version = getattr(module, attr, None)
-                # Cast any version specified as a list into a tuple.
-                version = tuple(version) if isinstance(version, list) else version
-                if version not in NULL_VERSIONS:
-                    return version
-            except Exception:
-                pass
-
-        # importlib was introduced into the standard library starting in Python3.8.
-        if "importlib" in sys.modules and hasattr(sys.modules["importlib"], "metadata"):
-            try:
-                version = sys.modules["importlib"].metadata.version(name)  # pylint: disable=E1101
-                if version not in NULL_VERSIONS:
-                    return version
-            except Exception:
-                pass
-
-        if "pkg_resources" in sys.modules:
-            try:
-                version = sys.modules["pkg_resources"].get_distribution(name).version
-                if version not in NULL_VERSIONS:
-                    return version
-            except Exception:
-                pass
-
     version = _get_package_version(name)
 
     # Coerce iterables into a string
     if isinstance(version, tuple):
         version = ".".join(str(v) for v in version)
 
     return version
+
+
+def get_package_version_tuple(name):
```

Comment:
This is perfect! I've been thinking about doing about this for a while but it was one of those low hanging fruit that I didn't have time for. Thanks for doing this! I wonder if it would be more immediately helpful to open a separate PR for this directly into main so we can take advantage of this immediately rather than it existing only on a feature branch though?

Reply:
I'd like to merge this in today and get it into the release so why don't we just ram this through rather than trying to separate?